Modules/Tokens/src/Tokens.Infrastructure/Persistence/Repository/TokensRepository.cs

```diff
@@ -29,19 +29,17 @@ public async Task<DbPaginationResult<Token>> FindTokens(IEnumerable<ListTokensQu
     {
         var queryItemsList = queryItems.ToList();
 
-        Expression<Func<Token, bool>> idAndPasswordFilter = template => false;
+        Expression<Func<Token, bool>> idFilter = template => false;
 
         foreach (var inputQuery in queryItemsList)
-        {
-            idAndPasswordFilter = idAndPasswordFilter
-                .Or(Token.HasId(TokenId.Parse(inputQuery.Id))
-                    .And(Token.CanBeCollectedWithPassword(activeIdentity, inputQuery.Password)));
-        }
+            idFilter = idFilter.Or(Token.HasId(TokenId.Parse(inputQuery.Id)));
 
         var query = (track ? _tokensDbSet : _readonlyTokensDbSet)
+            .IncludeAll(_dbContext)
             .Where(Token.IsNotExpired)
             .Where(Token.CanBeCollectedBy(activeIdentity))
-            .Where(idAndPasswordFilter);
+            .Where(Token.HasAllocationFor(activeIdentity))
+            .Where(idFilter);
```

Comment:
You should be able to simplify this query, now that we don't need to check the password anymore.

```c#
var query = (track ? _tokensDbSet : _readonlyTokensDbSet)
            .IncludeAll(_dbContext)
            .Where(Token.IsNotExpired)
            .Where(t => queryItems.Contains(t.Id)); // this line is new and replaces the foreach loop above)
```